Source/WebCore/platform/mediastream/gstreamer/GStreamerIncomingTrackProcessor.cpp

```diff
@@ -71,7 +71,7 @@ void GStreamerIncomingTrackProcessor::configure(ThreadSafeWeakPtr<GStreamerMedia
     if (auto ssrc = gstStructureGet<unsigned>(structure, "ssrc"_s)) {
         m_data.ssrc = *ssrc;
         auto msIdAttributeName = makeString("ssrc-"_s, *ssrc, "-msid"_s);
-        if (auto msIdAttribute = gstStructureGetString(structure, msIdAttributeName)) {
+        if (auto msIdAttribute = gstStructureGetString(structure, ASCIILiteral::fromLiteralUnsafe(msIdAttributeName.ascii().data()))) {
```

Comment:
Side point: This is incorrect. The passed-in characters are not a literal. So this is use after free.

Reply:
But according to the C++ spec the destructors will be called in reverse, right? If I am not mistaken the CString destructor of CString will be called after the destructor of ASCIILiteral and there would be not a UAF. Otherwise, all calls on String::ascii().data() would be doomed as well.

Reply:
And you made me wonder, so I wrote this piece of code which is a simplified but accurate version of what is going on here. Code is at https://pastebin.com/b3LVypDi and the result is what I suspected, there is no UAF because of the destructor order as per spec.

Reply:
And in case you wonder, I built it with clang and gcc, default and also `-O3`.

Reply:
Our goal with our safer C++ idiom is to be able to easily statically verify that the code doesn’t have certain classes of object lifetime problems. We don’t want to use `ASCIILiteral` with underlying string storage that isn’t “immortal”, even though we **could** do that successfully if we are careful about exactly what we do with those literals and with the lifetime of any `String` created from them. So as a matter of coding style we’d prefer not to do this unsafe optimization in the project.

Reply:
This should be solved it the new alias, I think.